config/metrics-server/deployment.yaml

```diff
@@ -72,6 +72,7 @@ spec:
           - mountPath: /apiserver.local.config/certificates
             name: certs
           securityContext:
+            runAsNonRoot: true
```

Comment:
I'm not sure if we should sneak this in this PR. Might be better to do in another PR + changelog entry?

Reply:
this is just a minor change because the policy is already defined at pod level, I added it to solve a semgrep issue that requires is also at container level